### PR TITLE
storageconsumer: modify api status for rbd, cephfs and cephuser

### PR DESCRIPTION
**Pull request by kesavanvt**
Modify StorageConsumer api to support status on node, provisioner user and subvolume group required for multi tenancy in cephfs.

Signed-off-by: kesavan kvellalo@redhat.com

### CODE REVIEW

File: api/v1alpha1/storageconsumer_types.go [hunk 1]
```diff
@@ -46,8 +45,12 @@ type StorageConsumerSpec struct {
 type CephObjectsSpec struct {
 	// BlockPoolName holds the name of created ceph block pool.
 	BlockPoolName string `json:"blockPoolName,omitempty"`
-	// CephUser holds the name of created ceph user.
-	CephUser string `json:"cephUser,omitempty"`
+	// CephRbdUser holds the name of created ceph user required for rbd.
+	CephRbdUser string `json:"cephRbdUser,omitempty"`
```
Comment (leseb) [inline api/v1alpha1/storageconsumer_types.go:49]: RBD is an acronym so it should be `CephRBDUser`, or else we can use `CephBlockPoolUser`.

File: api/v1alpha1/storageconsumer_types.go [hunk 2]
```diff
@@ -46,8 +45,12 @@ type StorageConsumerSpec struct {
 type CephObjectsSpec struct {
 	// BlockPoolName holds the name of created ceph block pool.
 	BlockPoolName string `json:"blockPoolName,omitempty"`
-	// CephUser holds the name of created ceph user.
-	CephUser string `json:"cephUser,omitempty"`
+	// CephRbdUser holds the name of created ceph user required for rbd.
```
Comment (leseb) [inline api/v1alpha1/storageconsumer_types.go:48]:
```suggestion
	// CephRbdUser holds the name of created ceph user required for rbd for the ceph-csi driver.
```

File: api/v1alpha1/storageconsumer_types.go [hunk 3]
```diff
@@ -46,8 +45,12 @@ type StorageConsumerSpec struct {
...
+	CephRbdUser string `json:"cephRbdUser,omitempty"`
+	// SubVolumeGroupName holds the name of created subVolume group.
+	SubVolumeGroupName string `json:"subVolumeGroupName,omitempty"`
+	// CephFsUser holds the name of created ceph user required for cephfs.
```
Comment (leseb) [inline api/v1alpha1/storageconsumer_types.go:52]:
```suggestion
	// CephFsUser holds the name of created ceph user required for cephfs for the ceph-csi driver.
```

File: api/v1alpha1/storageconsumer_types.go [hunk 4]
```diff
@@ -46,8 +45,12 @@ type StorageConsumerSpec struct {
 type CephObjectsSpec struct {
 	// BlockPoolName holds the name of created ceph block pool.
 	BlockPoolName string `json:"blockPoolName,omitempty"`
-	// CephUser holds the name of created ceph user.
-	CephUser string `json:"cephUser,omitempty"`
+	// CephRbdUser holds the name of created ceph user required for rbd.
+	CephRbdUser string `json:"cephRbdUser,omitempty"`
+	// SubVolumeGroupName holds the name of created subVolume group.
+	SubVolumeGroupName string `json:"subVolumeGroupName,omitempty"`
```
Comment (leseb) [inline api/v1alpha1/storageconsumer_types.go:51]: Aren't these fields mandatory? Why do we have `omitempty`?

Comment (kesavanvt) [reply api/v1alpha1/storageconsumer_types.go:51]: These fields are updated only when they are created and not to update before creation

File: api/v1alpha1/storageconsumer_types.go [hunk 5]
```diff
@@ -46,8 +45,12 @@ type StorageConsumerSpec struct {
...
 	// BlockPoolName holds the name of created ceph block pool.
 	BlockPoolName string `json:"blockPoolName,omitempty"`
-	// CephUser holds the name of created ceph user.
-	CephUser string `json:"cephUser,omitempty"`
+	// CephRBDUser holds the name of created ceph user required for rbd for the ceph-csi driver.
+	CephRBDUser string `json:"cephRBDUser,omitempty"`
+	// SubVolumeGroupName holds the name of created subVolume group.
+	SubVolumeGroupName string `json:"subVolumeGroupName,omitempty"`
+	// CephFsUser holds the name of created ceph user required for cephfs for the ceph-csi driver.
+	CephFsUser string `json:"cephFsUser,omitempty"`
```
Comment (umangachapagain) [inline api/v1alpha1/storageconsumer_types.go:53]:
```suggestion
	CephFSUser string `json:"cephFSUser,omitempty"`
```

File: api/v1alpha1/storageconsumer_types.go [hunk 6]
```diff
@@ -58,8 +61,6 @@ type StorageConsumerStatus struct {
 	GrantedCapacity resource.Quantity `json:"grantedCapacity,omitempty"`
 	// CephObjects provide details of ceph created objects required for external storage
 	CephObjects CephObjectsSpec `json:"cephObjects,omitempty"`
-	// ConnectionDetails holds the reference to secret containing external connection details
-	ConnectionDetails *v1.SecretKeySelector `json:"connectionDetails,omitempty"`
```
Comment (umangachapagain) [inline api/v1alpha1/storageconsumer_types.go:62]: This should probably be a separate PR or commit? Doesn't look related to adding new status fields.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

**Review by umangachapagain — CHANGES_REQUESTED** [review]
Put the generated changes into a single commit.

**Review by nb-ohad — APPROVED** [review]

**Review by iamniting — APPROVED** [review]
/lgtm
